fv3core/stencils/d2a2c_vect.py

```diff
@@ -41,26 +41,66 @@ def lagrange_interpolation_x_p1(qy: sd, qout: sd):
         qout = lagrange_x_func_p1(qy)
 
 
-@gtscript.function
-def avg_x(u):
-    return 0.5 * (u + u[0, 1, 0])
-
-
-@gtscript.function
-def avg_y(v):
-    return 0.5 * (v + v[1, 0, 0])
-
-
 @gtstencil()
 def avg_box(u: sd, v: sd, utmp: sd, vtmp: sd):
     with computation(PARALLEL), interval(...):
-        utmp = avg_x(u)
-        vtmp = avg_y(v)
+        utmp = 0.5 * (u + u[0, 1, 0])
+        vtmp = 0.5 * (v + v[1, 0, 0])
 
 
 @gtscript.function
-def contravariant(u, v, cosa, rsin):
-    return (u - v * cosa) * rsin
+def contravariant(v1, v2, cosa, rsin2):
```

Comment:
I assume we're switching `u, v` to `v1, v2` because `contravariant` works either way?

Reply:
Yes, and it is used both ways in this file.